microlith/loki/alerting/couchbase/couchbase-rules.yaml

```diff
@@ -236,3 +236,40 @@ groups:
           title: "Time Jumps Detected in memcached.log (Node: {{ $labels.hostname }})"
           description: Time jumps occur when the cluster manager process (`beam.smp`) has not been scheduled by the CPU, or not scheduled enough for a significant period of time.
           remediation: Check for evidence of your node being over provisioned or for evidence of VM resource contention.
+
+      - alert: CB90062-droppedTicks-instances # fires when over 4 instances of dropped ticks are detected
+        expr: |
+          count(count_over_time({file="debug.log"} |= "stats:warn" | json message="message" | line_format "{{.message}}" | pattern "<_> Dropped <ticks> ticks<_>" [1h])) > 4
```

Comment:
Im not sure how the rotation of debug.log is handled but if it doesn't happen very frequently then we could end up with false positives showing up. Would it be possible to also add a time limit to this expression e.g. 4 dropped ticks within x hours, or something akin to that?

Reply:
> Im not sure how the rotation of debug.log is handled

It's not :) IIRC Fluent Bit starts from wherever the file is when it starts and works forward from there (in other words, you'll only get logs _since_ FB started, nothing before then (*)), meaning it doesn't worry about rotation.

(*) unless you have restarts enabled - which, off the top of my head, our config does not

The query does have a `[1h]` in it though, so it should be fine?

Reply:
yeh the [1h] should cover this 👍